.github/workflows/test.yml

```
@@ -7,35 +7,8 @@ jobs:
     name: Testing Biotite without time consuming test cases
     strategy:
       matrix:
-        include:
-          - os: ubuntu-latest
-            python: '3.8'
-            numpy: '1.16'
-          - os: ubuntu-latest
-            python: '3.9'
-            numpy: '1.19'
-          - os: ubuntu-latest
-            python: '3.10'
-            numpy: '1.21'
-          - os: macos-latest
-            python: '3.8'
-            numpy: '1.16'
-          - os: macos-latest
-            python: '3.9'
-            numpy: '1.19'
-          - os: macos-latest
-            python: '3.10'
-            numpy: '1.21'
-          - os: windows-latest
-            python: '3.8'
-            numpy: '1.16'
-          - os: windows-latest
-            python: '3.9'
-            numpy: '1.19'
-          - os: windows-latest
-            python: '3.10'
-            numpy: '1.21'
-
+        os: ['ubuntu-latest', 'windows-latest', 'macos-latest']
+        pyrestrictions: [{python: '3.8', numpy: '1.19'}, {python: '3.10', numpy: '1.21'}, {python: '3.11', numpy: '1.23'}]
```

I like the organization using the Cartesian product of OS and version, it looks much cleaner to me. However I think the key name `versions` would be more descriptive and shorter than `pyrestrictions`. Furthermore I think you wanted to write Python 3.9 rather than 3.8.
```suggestion
        versions: [{python: '3.9', numpy: '1.19'}, {python: '3.10', numpy: '1.21'}, {python: '3.11', numpy: '1.23'}]
```

---

I just did 3.8 because that was the minimum version you had tested previously. I'll update it though.